test/conftest.py

```diff
@@ -48,6 +53,9 @@ def get(self, url: str, benchmark: Optional[Callable] = None) -> Response:
 
         return response
 
+    def assert_404(self, url: str):
+        raise NotImplementedError()
```

Comment:
Error handling is unfortunately server-specific which is why I handle it as a method of `BaseServer`